lib/chef/resource/windows_user_privilege.rb

```diff
@@ -132,8 +141,8 @@ class WindowsUserPrivilege < Chef::Resource
          }
 
       load_current_value do |new_resource|
-        unless new_resource.principal.nil?
-          privilege Chef::ReservedNames::Win32::Security.get_account_right(new_resource.principal) unless new_resource.action.include?(:set)
+        unless new_resource.principal.nil? || new_resource.action.include?(:set) || new_resource.action.include?(:clear)
```

Comment:
I know this is not all new code you're adding but I think this would be better with the logic reversed:

```suggestion
        if new_resource.principal && (new_resource.action.include?(:add) || new_resource.action.include?(:remove)
```

Reply:
@phiggins thanks for the input. suggestion committed.